hipe4ml/TrainingAndTesting/generalized_analysis.py

```diff
@@ -0,0 +1,131 @@
+'''Module containing the class to perform the training and the testing of the models '''
+
+
+import pandas as pd
+import uproot
+from sklearn.model_selection import train_test_split
+
+
+class GeneralizedAnalysis:
```

Comment:
I would generalise the class such that the variables (pT, cT, centrality) are passed as one or more given column of the data frames